src/common/redis_module/ray_redis_module.cc

```diff
@@ -507,6 +563,29 @@ int TableAdd_RedisCommand(RedisModuleCtx *ctx,
     if (reply == NULL) {
       RedisModule_ReplyWithError(ctx, "error during PUBLISH");
     }
+
+    // Publish the data to any clients who requested notifications on this key.
+    RedisModuleKey *notification_key = OpenBroadcastKey(
+        ctx, pubsub_channel_str, id, REDISMODULE_READ | REDISMODULE_WRITE);
+    if (RedisModule_KeyType(notification_key) != REDISMODULE_KEYTYPE_EMPTY) {
+      // NOTE(swang): Sets are not implemented yet, so we use ZSETs instead.
+      CHECK_ERROR(RedisModule_ZsetFirstInScoreRange(
+                      notification_key, REDISMODULE_NEGATIVE_INFINITE,
+                      REDISMODULE_POSITIVE_INFINITE, 1, 1),
+                  "Unable to initialize zset iterator");
+      for (; !RedisModule_ZsetRangeEndReached(notification_key);
+           RedisModule_ZsetRangeNext(notification_key)) {
+        RedisModuleString *client_channel =
+            RedisModule_ZsetRangeCurrentElement(notification_key, NULL);
+        RedisModuleCallReply *reply =
+            RedisModule_Call(ctx, "PUBLISH", "ss", client_channel, data);
+        if (reply == NULL) {
+          RedisModule_CloseKey(notification_key);
+          RedisModule_ReplyWithError(ctx, "error during PUBLISH");
```

Comment:
Should this be `return RedisModule_ReplyWithError(ctx, "error during PUBLISH");`?  Otherwise the key will be closed twice (which I think is fine, just somewhat stands out as odd).

Reply:
Thanks!